datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -1324,6 +1325,25 @@ pub fn change_redundant_column(fields: &Fields) -> Vec<Field> {
         })
         .collect()
 }
+
+fn mark_field(schema: &DFSchema) -> (Option<TableReference>, Arc<Field>) {
+    let mut table_references = schema
+        .iter()
+        .filter_map(|(qualifier, _)| qualifier)
+        .collect::<Vec<_>>();
+    table_references.dedup();
+    let table_reference = if table_references.len() == 1 {
+        table_references.first().cloned().cloned()
```

Comment:
Is this doing double clone?

Is `table_references[0].clone()` better?

Reply:
I because the type is `Option<&&TableReference>` will change it to `pop()` instead of `first()` to avoid one of the `cloned`.